### DISCUSSION TITLE
Auto generate access token

### DISCUSSION THREAD

**Discussion by mike-serchenia** [argoproj/argo-cd#10584] [Q&A]
Hi

is there a way to autogenerate user access token?

for example instead of running manually `argocd account generate-token --account autouser` we can set some parameter during installation and have that token accessible from k8s secret

**Comment by ryanwuer** [comment]
+1

**Comment by mfabricanti** [comment]
+1

**Comment by mfabricanti** [comment]
Is there a way to manually insert (Declarative way) the token value in one account?

When I generate a token (from UI), in Secret argocd-secret appears only:
**accounts.<account_name>.tokens: [{"id":"cbbc0bdc-...<redacted>","iat":1692903744}]**

**Comment by mfabricanti** [comment]
After studying ArgoCD's code a little bit, I was able to figure out how this mechanism works.

Once you generate a token for an account, in any Argo instance, all you have to do, to make this token works in another Argo instance (another cluster, for example) is to have the secret argocd-secret with the same values like below:
- accounts.<account_name>.tokens
- server.secretkey

**Comment by ashish1099** [comment]
you can use the helm chart values.yaml to add it
